SportsHub/SportsHub.Api/Exceptions/ExceptionHandler.cs

```diff
@@ -1,20 +1,55 @@
 ﻿using System.Net;
+using SportsHub.Api.Exceptions.CustomExceptionModels;
 
 namespace SportsHub.Api.Exceptions;
 
 public class ExceptionHandler: IMiddleware
 {
+    private CustomException _exception;
     public async Task InvokeAsync(HttpContext context, RequestDelegate next)
     {
         try
         {
             await next(context);
         }
-
-        catch (Exception e)
+        catch (BusinessLogicException ex)
+        {
+            await HandleExceptionAsync(context, ex);
+        }
+        catch (NotFoundException ex)
+        {
+            await HandleExceptionAsync(context, ex);
+        }
+        catch (Exception ex)
         {
             context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
-            await context.Response.WriteAsync(e.Message);
+            await context.Response.WriteAsync(ex.Message);
         }
     }
+
+    private async Task HandleExceptionAsync(HttpContext context, Exception ex)
+    {
+        string result = null;
+        context.Response.ContentType = "application/json";
+        
+        switch (ex)
+        {
+            case BusinessLogicException:
+            {
+                _exception = new BusinessLogicException("There has been error in you business logic");
```

Comment:
is this switch for the test purpose only?
in the InvokeAsync you call this method for a different exception. and here again, you have to check for the exception type. Make this method generic, please.

Reply:
Yea it was for testing purpose only . The idea of was seeing some basic concept of what my idea is . I will make it Generic or i can make it depend on abstraction